### ISSUE TITLE
Exposing the console

### ISSUE THREAD

**Issue by Ramblurr** [laverdet/screepsmod-inspector#1]
This is awesome @laverdet, and it works nicely.

Any idea how to go about exposing a 'real' V8 console to the devtools intance?

Inside the ivm, `global.console` is obviously the fake console from screeps engine.

Could we expose it as a different object `global.debugConsole` perhaps? I'd like to set a debug flag in my code to log to this console during debugger sessions.

**Comment by laverdet** [comment]
Well, v8 doesn't actually have a console. `console` is a whatwg specification and v8 only implements javascript. I just pushed (3a87c11f9b6f0da604da455b30820ec054f21539) my `print` function that I've been using on my end. That'll output strings to stdout on the runner.

**Comment by Ramblurr** [comment]
Ah interesting. I don't know much about the v8 internals.

I tried something similar to your commit yesterday, essentially passing a console.log wrapper function to the ivm.

However this resulted in the output being sent to the node runner's stdout, which was looking gged to the log file.

Ideally, the output from the ivm would appear in the console tab of chrome devtools...rather than the stdout of the runner process.

I ran the runner process with --inspect, which let's me attach to it and see the console output there..but then the user experience of debugging a script is split between the runner's inspector where consioe output goes and the ivm inspector where debugging the code happens.

**Comment by Ramblurr** [comment]
I checked out isolated-vm and ran the [inspector-example.js](https://github.com/laverdet/isolated-vm/blob/master/inspector-example.js) test, but I edited the infinite loop to also have a console.log statement.

![](https://user-images.githubusercontent.com/14830/39634522-5584e9b0-4fbb-11e8-89f3-6daea74aa227.png)

So this is definitely possible already in terms of isolated-vm, it is just a matter of getting `screepsmod-inspector` to (re)wire up the context to expose the "original" console.

**Comment by Ramblurr** [comment]
(**edit** I deleted an older comment after much diving through the codebase, I understand the architecture better now, and my old comments were just noise.)

@laverdet I've pulled your change with the `print` function. But the *inner* async function is never called 
(verified with a print statement before and after). 

`isolated-vm` should be added to the deps list of this mod.

Also if I understand it correctly, you're wrapping the console instance of the runner process. Not the console instance (clobbered [here](https://github.com/screeps/engine/blob/isolated-vm/src/game/game.js#L398))inside the ivm which would be displayed in the dev tools console. So `print()` would output to the sever logfile not devtools. This would achieve what I'm aiming for:

```js
await (await this.compileScript('realConsole = console')).run(context);
```
